From 3906b2b60881f34d8c9ee7aeb73786f003425690 Mon Sep 17 00:00:00 2001
From: Masami Ichikawa <masami.ichikawa@miraclelinux.com>
Date: Fri, 25 Jun 2021 10:39:29 +0900
Subject: [PATCH 1/3] Apply patch:
 0001-linux-user-assume-__NR_gettid-always-exists.patch

Backport patch 0001-linux-user-assume-__NR_gettid-always-exists.patch[1]
from poky thud branch.

This commit fixed hunk.

1:
https://git.yoctoproject.org/cgit/cgit.cgi/poky/patch/meta/recipes-devtools/qemu/qemu/0001-linux-user-assume-__NR_gettid-always-exists.patch?id=f3a4b208500790f2163dab3fefc8f18b57d37621

Signed-off-by: Masami Ichikawa <masami.ichikawa@miraclelinux.com>
---
 linux-user/syscall.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index efdd000..8c2e8ec 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -249,15 +249,7 @@ static type name (type1 arg1,type2 arg2,type3 arg3,type4 arg4,type5 arg5,	\
 #define TARGET_NR__llseek TARGET_NR_llseek
 #endif
 
-#ifdef __NR_gettid
 _syscall0(int, gettid)
-#else
-/* This is a replacement for the host gettid() and must return a host
-   errno. */
-static int gettid(void) {
-    return -ENOSYS;
-}
-#endif
 
 /* For the 64-bit guest on 32-bit host case we must emulate
  * getdents using getdents64, because otherwise the host
-- 
2.25.1

